src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -713,18 +713,40 @@ public static unsafe double Round(double value, int digits, MidpointRounding mod
 
                 value *= power10;
 
-                if (mode == MidpointRounding.AwayFromZero)
+                if (mode is MidpointRounding.ToEven)
                 {
-                    var fraction = ModF(value, &value);
+                    value = Round(value);
+                }
+                else if (mode is MidpointRounding.AwayFromZero)
+                {
+                    double fraction = ModF(value, &value);
 
                     if (Abs(fraction) >= 0.5)
                     {
                         value += Sign(fraction);
                     }
                 }
-                else
+                else if (mode is MidpointRounding.ToZero)
                 {
-                    value = Round(value);
+                    value = Truncate(value);
+                }
+                else if (mode is MidpointRounding.ToNegativeInfinity)
+                {
+                    double fraction = ModF(value, &value);
+
+                    if (Sign(fraction) == -1)
```

Comment:
This should only come into play for ties.

Reply:
Decimal.DecCalc.InternalRound with Truncate mode supplied (AKA ToZero if i'm not mistaken) currently behaves like this aswell.

e.g. these scenarios all pass..
```cs
[InlineData(11.0, 11, MidpointRounding.ToZero)]
[InlineData(11.4, 11, MidpointRounding.ToZero)]
[InlineData(11.5, 11, MidpointRounding.ToZero)]
[InlineData(11.6, 11, MidpointRounding.ToZero)]
[InlineData(-11.0, -11, MidpointRounding.ToZero)]
[InlineData(-11.4, -11, MidpointRounding.ToZero)]
[InlineData(-11.5, -11, MidpointRounding.ToZero)]
[InlineData(-11.6, -11, MidpointRounding.ToZero)]
public static void Round_Decimal_Modes(decimal x, decimal expected, MidpointRounding mode)
{
    Assert.Equal(expected, Math.Round(x, 0, mode));
    Assert.Equal(expected, decimal.Round(x, 0, mode));       
}
```
Agreed its not really midpoint rounding though, perhaps Math.Round(decimal, ..) could be implemented in Math.cs ?

Reply:
Right. Sorry, I was thinking about the case where you are parsing a result.

Having comments explaining each case would be helpful. We should also just be able to use `Ceiling`, `Floor`, and `Truncate` (for `ToPositiveInfinity`, `ToNegativeInfinity`, and `Truncate`, respectively)

Reply:
> 
> 
> Right. Sorry, I was thinking about the case where you are parsing a result.
> 
> Having comments explaining each case would be helpful. We should also just be able to use `Ceiling`, `Floor`, and `Truncate` (for `ToPositiveInfinity`, `ToNegativeInfinity`, and `Truncate`, respectively)

Done, added some comments and now using Floor, Ceiling and Truncate.